src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/GivenADependencyContextBuilder.cs

```diff
@@ -211,6 +213,168 @@ public void ItHandlesReferenceAndPackageReferenceNameCollisions()
                 .Contain(c => c.Name == "System.Collections.NonGeneric.Reference.Reference" && c.Type == "referenceassembly");
         }
 
+        // If an assembly is in withResources, it has to be a key in dependencies, even with an empty list.
+        private static DependencyContext BuildDependencyContextFromDependenciesWithResources(Dictionary<string, List<string>> dependencies, List<string> withResources, List<string> references)
+        {
+            string mainProjectName = "simpleApp";
+            LockFile lockFile = TestLockFiles.GetLockFile(mainProjectName);
+
+            SingleProjectInfo mainProject = SingleProjectInfo.Create(
+                "/usr/Path",
+                mainProjectName,
+                ".dll",
+                "1.0.0",
+                []);
+            string mainProjectDirectory = Path.GetDirectoryName(mainProject.ProjectPath);
+
+            ITaskItem[] referencePaths = references.Select(reference =>
+                new MockTaskItem($"/usr/Path/{reference}.dll", new Dictionary<string, string> {
+                    { "CopyLocal", "false" },
+                    { "FusionName", $"{reference}, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null" },
+                    { "Version", "" },
+                })).ToArray();
+
+            ProjectContext projectContext = lockFile.CreateProjectContext(
+                FrameworkConstants.CommonFrameworks.NetCoreApp10.GetShortFolderName(),
+                runtime: null,
+                platformLibraryName: Constants.DefaultPlatformLibrary,
+                runtimeFrameworks: null,
+                isSelfContained: false);
+
+            Dictionary<string, SingleProjectInfo> referenceProjectInfos = new();
+
+            foreach (KeyValuePair<string, List<string>> kvp in dependencies)
+            {
+                projectContext.LockFileTarget.Libraries.Add(
+                    new LockFileTargetLibrary()
+                    {
+                        Name = kvp.Key,
+                        Version = new NuGetVersion(4, 0, 0),
+                        Type = withResources.Contains(kvp.Key) ? "project" : "unrealType",
+                        Dependencies = kvp.Value.Select(n => new PackageDependency(n)).ToList()
+                    });
+
+                if (withResources.Contains(kvp.Key))
```

Comment:
I think having resource assets is not the most common case, so instead we should probably model whether the libraries have runtime or compilation assets (and then resource assets if we have additional scenarios for that).

Reply:
All three are handled in the same way in this part of the code, but they're added in different ways...I don't disagree with you, exactly, but I suspect it would take me a couple days to get that to work, and it wouldn't actually test anything we aren't already properly testing using resource assets.